src/components/file/FileUploader.js

```diff
@@ -48,53 +40,20 @@ const useStyles = makeStyles((theme) => ({
 
 const FileUploader = () => {
   const classes = useStyles();
-  const match = useMatch(buildItemPath());
-  const itemId = match?.params?.itemId;
-  const [uppy, setUppy] = useState(null);
+  const { uppy } = useContext(UppyContext);
   const [isDragging, setIsDragging] = useState(false);
   const [isValid, setIsValid] = useState(true);
-  const { mutate: onFileUploadComplete } = useMutation(
-    MUTATION_KEYS.FILE_UPLOAD,
-  );
   const { t } = useTranslation();
 
   const closeUploader = () => {
     setIsDragging(false);
   };
 
-  const onFilesAdded = () => {
-    closeUploader();
-  };
-
-  const onComplete = (result) => {
-    // update app on complete
-    // todo: improve with websockets or by receiving corresponding items
-    if (result?.successful?.length) {
-      onFileUploadComplete({ id: itemId });
-    }
-
-    return false;
-  };
-
-  const onError = (error) => {
-    onFileUploadComplete({ id: itemId, error });
-  };
-
-  const onUpload = () => {
-    notifier({ type: uploadFileRoutine.REQUEST });
-  };
-
-  const applyUppy = () => {
-    setUppy(
-      configureFileUppy({
-        itemId,
-        onComplete,
-        onFilesAdded,
-        onError,
-        onUpload,
-      }),
-    );
-  };
+  useEffect(() => {
+    uppy?.on('files-added', () => {
```

Comment:
Won't `uppy` always have the `on`? Or is this checking if `uppy` is not falsy?

Reply:
Uppy could be not configured yet.